react/w-books/src/app/screens/SignUp/components/SignUpForm/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ChangeEvent, Component } from 'react';
+
+import Form from '../../../../components/Form';
+import FormRow from '../../../../components/FormRow';
+import Button from '../../../../components/Button';
+
+import styles from './styles.module.scss';
+
+interface SignUpFormState {
+  formData: UserData;
+}
+
+interface UserData {
+  email: string;
+  password: string;
+  password_confirmation: string;
+  first_name: string;
+  last_name: string;
+  locale: string;
+}
+export default class SignUpForm extends Component<{}, SignUpFormState> {
+  handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (event?.target?.id) {
+      event.persist();
+
+      this.setState((actualState) => (
+        {
+          formData: { ...actualState?.formData, [event.target.id]: event.target.value }
+        }
+      ));
+    }
+  }
+
+  signUp = () => {
+    if (this.state && this.state.formData) {
+      const data = { ...this.state.formData, locale: navigator.language };
+
+      console.log(data);
+    }
+  }
+
+  render() {
+    return (
+      <Form className={styles['sign-up-form']} handleSubmit={this.signUp}>
+        <FormRow
+          labelName="Nombre"
+          className={styles['form-row']}
+          inputProps={{
+            id: 'first_name',
+            handleChange: this.handleInputChange,
+            type: 'text'
+          }}
```

Comment:
this prop looks good, but in each render, you are sending a new object with the same props in something cases, i think that is better send separately this props